
Fix broken snd_pcm_hw_params_get_channels useage ...

  - Rene Rebe <rene@rocklinux.org>

--- SDL-1.2.6/src/audio/alsa/SDL_alsa_audio.c.orig	2004-01-11 20:16:20.000000000 +0100
+++ SDL-1.2.6/src/audio/alsa/SDL_alsa_audio.c	2004-01-11 20:17:30.000000000 +0100
@@ -269,8 +269,9 @@
 	/* Set the number of channels */
 	status = snd_pcm_hw_params_set_channels(pcm_handle, params, spec->channels);
 	if ( status < 0 ) {
-		status = snd_pcm_hw_params_get_channels(params);
-		if ( (status <= 0) || (status > 2) ) {
+		int ch;
+		status = snd_pcm_hw_params_get_channels(params, &ch);
+		if ( (status < 0) || (ch != spec->channels) ) {
 			SDL_SetError("Couldn't set audio channels");
 			ALSA_CloseAudio(this);
 			return(-1);
